### PR TITLE
Use secure randoms where needed

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Use secure random number generators in Java code.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [X] **DONE**

## Test coverage
- No tests: already covered

- [X] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/21411

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by mackdk — APPROVED** [review]

**Comment by cbosdo** [comment]
I still don't know what to do with https://sonarcloud.io/project/security_hotspots?id=uyuni-project_uyuni&hotspots=AYMYhYO-5TQr2KFg-0DZ generating md5... yuk!

File: java/code/src/com/redhat/rhn/manager/system/SystemManager.java [hunk 1]
```diff
@@ -3933,7 +3934,7 @@ public static void setReportDbUser(MinionServer minion, boolean forcePwChange) {
         }
         Credentials credentials = Optional.ofNullable(mgrServerInfo.getReportDbCredentials())
                 .orElse(CredentialsFactory.createCredentials(
-                        "hermes_" + RandomStringUtils.random(8, "abcdefghijklmnopqrstuvwxyz"),
+                        "hermes_" + RandomStringUtils.random(8, 0, 0, true, false, null, new SecureRandom()),
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/manager/system/SystemManager.java:3937]: Not sure about this. We just want lowercase characters. Security is not important here as we generate a (unique) username and not a password.

Comment (cbosdo) [reply java/code/src/com/redhat/rhn/manager/system/SystemManager.java:3937]: I can lowercase the generated result... or ignore that warning, indeed

Comment (cbosdo) [reply java/code/src/com/redhat/rhn/manager/system/SystemManager.java:3937]: oops, I forgot that I still needed to fix this one... I'll prepare a follow up PR

**Pull request merged**